File: ios/ZhengQi/Models/CheckIn.swift
import Foundation
import SwiftData

@Model
final class CheckIn {
    var id: UUID
    var trackItem: TrackItem?
    var date: String
    var status: Bool
    var note: String
    var imageData: Data?
    var createdAt: Date

    init(id: UUID = UUID(), trackItem: TrackItem? = nil, date: String = "", status: Bool = false, note: String = "", imageData: Data? = nil, createdAt: Date = Date()) {
        self.id = id
        self.trackItem = trackItem
        self.date = date
        self.status = status
        self.note = note
        self.imageData = imageData
        self.createdAt = createdAt
    }
}

File: android/core/src/main/java/com/zhiguan/app/data/model/CheckIn.kt
package com.zhiguan.app.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "check_ins")
data class CheckIn(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val trackItemId: Long,
    val date: String,
    val status: Boolean,
    val note: String = "",
    val imageUri: String = "",
    val createdAt: Long = System.currentTimeMillis()
)